src/components/AddColumnBtn/CustomColumnModal.js

```diff
@@ -0,0 +1,958 @@
+import React from 'react'
+import { v4 as uuid } from 'uuid'
+import PropTypes from 'prop-types'
+import _cloneDeep from 'lodash.clonedeep'
+
+import {
+  ColumnObj,
+  deepEqual,
+  ColumnTypes,
+  COLUMN_TYPES,
+  getVisibleColumns,
+  formatQueryColumns,
+  isColumnNumberType,
+  autoQLConfigDefault,
+  authenticationDefault,
+  dataFormattingDefault,
+  getColumnTypeAmounts,
+  isColumnStringType,
+} from 'autoql-fe-utils'
+
+import { Icon } from '../Icon'
+import { Modal } from '../Modal'
+import { Input } from '../Input'
+import { Button } from '../Button'
+import { Select } from '../Select'
+import ChataTable from '../ChataTable/ChataTable'
+import { ErrorBoundary } from '../../containers/ErrorHOC'
+import { authenticationType, autoQLConfigType, dataFormattingType } from '../../props/types'
+import {
+  createMutatorFn,
+  getFnSummary,
+  WINDOW_FUNCTIONS,
+  ORDERABLE_WINDOW_FN_TYPES,
+  getOperators,
+} from './customColumnHelpers'
+
+import './CustomColumnModal.scss'
+
+const globalOperators = ['LEFT_BRACKET', 'RIGHT_BRACKET']
+const FUNCTION_OPERATOR = 'FUNCTION'
+
+const HIGHLIGHTED_CLASS = 'highlighted-column'
+const DISABLED_CLASS = 'disabled-column'
+const FORMULA_CLASS = 'formula-column'
+const DEFAULT_COLUMN_NAME = 'New Column'
+
+export const getSelectableColumns = (columns) => {
```

Comment:
Ohh yes I actually did this intentionally, because the getSelectableColumns function will eventually include more than just numerical columns as we add to the feature. So just future proofing